src/plugins/catalog/store.go

```diff
@@ -41,3 +45,21 @@ func (catalogApi *CatalogApi) Get(catalogId uint64) (Catalog, error) {
 func (catalogApi *CatalogApi) Delete(catalogId uint64) error {
 	return catalogApi.DbClient.Delete(&Catalog{ID: catalogId}).Error
 }
+
+func (catalogApi *CatalogApi) Update(catalog *Catalog) error {
+	count := 0
+	if err := catalogApi.DbClient.
```

Comment:
how about extract to a function with code line#10 - line#15, 
catalogCountWithName(catalogName)